Remove the `rust` scenario from GH Actions testing

jsf9k opened the pull request:
## 🗣 Description ##

This pull request removes the `rust` scenario from GitHub Actions testing.

## 💭 Motivation and context ##

[Porchetta-Industries/CrackMapExec](https://github.com/Porchetta-Industries/CrackMapExec) is no longer being installed in our Kali AMI as of cisagov/kali-packer#182, so we no longer have an example of a tool we are installing that requires Rust.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Post-approval checklist ##

- [x] Remove `rust` checks as required for this repo.

jsf9k committed:
Remove the Rust scenario from GH Actions testing

Porchetta-Industries/CrackMapExec is no longer being installed in our
Kali AMI as of cisagov/kali-packer#182, so we no longer have an
example of a Rust installation.

dav3r reviewed (approved):
Rust in peace!

mcdonnnj reviewed (approved):
My only suggestion is to add an issue to restore testing for Rust tools when possible. Otherwise this LGTM ✔️

jsf9k committed:
Add reference to issue

GH Actions testing of the rust Molecule scenario should be re-enabled
when we again have an example of an assessment tool that requires
Rust.  See #64 for more details.

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

jsf9k commented:
> My only suggestion is to add an issue to restore testing for Rust tools when possible. Otherwise this LGTM ✔️

Please see #64 and commit b971eb6.
